v0/destinations/zendesk/transform.js

```diff
@@ -453,33 +473,21 @@ async function processIdentify(message, destinationConfig, headers) {
   );
   const url = endPoint + category.createOrUpdateUserEndpoint;
   const returnList = [];
+  let userId;
 
-  const userId = await getUserId(message, headers);
-  // If primaryEmail and userId both exists then it will set it as primary email.
-  if (primaryEmail && userId) {
-    const userIdentityId = await getUserIdentityId(
-      userId,
-      primaryEmail,
-      headers,
-      destinationConfig
-    );
-    if (userIdentityId) {
-      returnList.push(
-        responseBuilderToSetPrimaryAccount(userIdentityId, userId, headers)
+  if (destinationConfig.searchByExternalId) {
+    userId = await getUserIdByExternalId(message, headers, traits);
+    if (userId) {
+      const payloadForUpdatingEmail = await payloadBuilderforUpdatingEmail(
+        message,
+        userId,
+        headers
       );
+      if (payloadForUpdatingEmail && !isEmptyObject(payloadForUpdatingEmail))
+        returnList.push(payloadForUpdatingEmail);
     }
-  } else if (primaryEmail) {
-    // if only primaryEmail exists then in the payload it will pass as user.identities
-    // and it will set it as primary while creating the user.
-    const primaryEmailPayload = {
-      type: "email",
-      value: primaryEmail
-    };
-    if (payload.user.identities && Array.isArray(payload.user.identities)) {
-      payload.user.identities.push(primaryEmailPayload);
-    } else {
-      payload.user.identities = [primaryEmailPayload];
-    }
+  } else {
+    userId = await getUserId(message, headers);
```

Comment:
We can move this to the previous section I mean at line number 499, otherwise, it will increase one get call for some previous users.

Reply:
done